test/Base_Tests/src/Data/Vector_Spec.enso

```diff
@@ -778,21 +776,19 @@ type_spec suite_builder name alter = suite_builder.group name group_builder->
 
     group_builder.specify "should correctly propagate state through each" <|
         v = State.run Number 77 <|
-            b = Vector.new_builder
-            (alter ["A", "B"]).each x->
-                b.append x
-                b.append (State.get Number).to_text
-            b.to_vector
+            Vector.build builder->
```

Comment:
I'd assume this is a general even for complicated computations. Just wrap your computation in `Vector.build builder->` block and that's it.